src/bindings/jna/libelektra/src/main/java/org/libelektra/Key.java

```diff
@@ -1,34 +1,113 @@
+import static org.libelektra.Key.KeyNewArgumentTag.KEY_VALUE;
+import static org.libelektra.ValidationUtil.argNotNull;
+import static org.libelektra.ValidationUtil.argNotNullOrBlank;
+import static org.libelektra.ValidationUtil.checkKeyPointer;
 
 import com.sun.jna.Pointer;
 import java.lang.ref.Cleaner;
 import java.util.Arrays;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
 import java.util.Optional;
+import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
-import org.libelektra.Elektra.KeyNewArgumentFlags;
 import org.libelektra.exception.KeyBinaryTypeNotSupportedException;
-import org.libelektra.exception.KeyCreateFailedException;
+import org.libelektra.exception.KeyException;
+import org.libelektra.exception.KeyMetaException;
+import org.libelektra.exception.KeyNameException;
 import org.libelektra.exception.KeyReleasedException;
-import org.libelektra.exception.KeySetNameFailedException;
 import org.libelektra.exception.PluginMisbehaviorException;
 
 /**
- * Key is an essential class that encapsulates key name , value and metainfo.
+ * Key represents an native Elektra key providing access to its name, value and
+ * meta information
  */
 public class Key implements Iterable<String>
 {
 
 	private static final String WARNINGS = "warnings";
 
-	// constants
+	/**
+	 * Argument tags for use with {@link #create(String, Object...)}
+	 */
+	public enum KeyNewArgumentTag {
+
+		/**
+		 * Used as a parameter terminator
+		 */
+		KEY_END (0),
+
+		/**
+		 * Flag for the key name
+		 */
+		KEY_NAME (1),
+
+		/**
+		 * Flag for the key data
+		 */
+		KEY_VALUE (1 << 1),
+
+		/**
+		 * Flag for the key comment
+		 */
+		KEY_COMMENT (1 << 3),
+
+		/**
+		 * Flag if the key is binary
+		 */
+		KEY_BINARY (1 << 4),
+
+		/**
+		 * Flag for maximum size to limit value
+		 */
+		KEY_SIZE (1 << 11),
+
+		/**
+		 * Flag for metadata
+		 */
+		KEY_META (1 << 15);
+
+		public final Integer value;
+
+		private KeyNewArgumentTag (int value)
+		{
+			this.value = Integer.valueOf (value);
+		}
+	}
+
+	/**
+	 * Flag for use with {@link #copy(Key, int)} for copying the key name
+	 */
 	public static final int KEY_CP_NAME = 1 << 0;
+
+	/**
+	 * Flag for use with {@link #copy(Key, int)} for copying the key value, if it is
+	 * a string
+	 *
+	 * @apiNote Do not use together with {@link #KEY_CP_VALUE}
+	 */
 	public static final int KEY_CP_STRING = 1 << 1;
+
+	/**
+	 * Flag for use with {@link #copy(Key, int)} for copying the key value
+	 *
+	 * @apiNote Do not use together with {@link #KEY_CP_STRING}
+	 */
 	public static final int KEY_CP_VALUE = 1 << 2;
+
+	/**
+	 * Flag for use with {@link #copy(Key, int)} for copying the key metadata
+	 */
 	public static final int KEY_CP_META = 1 << 3;
+
+	/**
+	 * Flag for use with {@link #copy(Key, int)} for copying the key name, value and
+	 * metadata
+	 */
 	public static final int KEY_CP_ALL = KEY_CP_NAME | KEY_CP_VALUE | KEY_CP_META;
```

Comment:
i thought about that and tested how the api would feel with these constants in an enum. Since these constants are intended to be used as combinable flags/bitmask, the API syntax would be more complicated than necessary: to get the flags on would have to append some member name to get the value.
In contrary, Key.NewArgumentTag is not meant to combined with other values and therefor the API can work with the enum itself.